### PR TITLE
ddt removal in api user and role tests

### PR DESCRIPTION
**Pull request by sthirugn**
user tests:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 5 tests in 112.880s

OK
```

role tests:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 3 tests in 10.469s

OK
```

### CODE REVIEW

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -16,15 +15,13 @@
...
 from robottelo.test import APITestCase
 
 
-@ddt.ddt
-class RoleTestCase(APITestCase):
-    """Tests for ``api/v2/roles``."""
-
-    @decorators.data(
+def data_generator():
+    """Returns data for role name"""
```
Comment (elyezer) [inline tests/foreman/api/test_role.py]: Return a tuple containing data generators function to be used to generate role name?

File: tests/foreman/api/test_role.py [hunk 2]
```diff
@@ -16,15 +15,13 @@
...
+from robottelo.decorators import bz_bug_is_open
 from robottelo.test import APITestCase
 
 
-@ddt.ddt
-class RoleTestCase(APITestCase):
-    """Tests for ``api/v2/roles``."""
-
-    @decorators.data(
+def data_generator():
```
Comment (elyezer) [inline tests/foreman/api/test_role.py]: s/data_generator/data_generators?

File: tests/foreman/api/test_user.py [hunk 3]
```diff
@@ -4,18 +4,21 @@
...
 from nailgun import entities
 from random import randint
 from requests.exceptions import HTTPError
-from robottelo import decorators
 from robottelo.test import APITestCase
 
 
 def _user_attrs():
-    """Return an iterable of dicts. Each dict contains user attributes."""
+    """Return an iterable of dicts. Each dict contains user attributes.
```
Comment (elyezer) [inline tests/foreman/api/test_user.py]: s/an iterable/tuple

**Comment by elyezer** [comment]
ACK, comments not blockers.

File: tests/foreman/api/test_role.py (hunk 2 above)
Comment (sthirugn) [inline tests/foreman/api/test_role.py]: Here `data` is plural and `generator` is singular.

This is a single generator returning different types of data.

Comment (elyezer) [inline tests/foreman/api/test_role.py]: Got it.

**Comment by sthirugn** [comment]
Thank you for the comments, updated now.

**Comment by sthirugn** [comment]
Rebased and ready for final review/merge

**Comment by elyezer** [comment]
ACK

File: tests/foreman/api/test_role.py (hunk 2 above)
Comment (JacobCallahan) [inline tests/foreman/api/test_role.py]: English is weird at times :)

Comment (sthirugn) [inline tests/foreman/api/test_role.py]: yeah ha ha :)

**Comment by omaciel** [comment]
ACK

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**